test/TestUtilities/Test.Utility/TestPackageSpecFactory.cs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Commands.Restore;
+using NuGet.Commands.Restore.Utility;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.ProjectModel;
+
+namespace Test.Utility;
+
+public class TestPackageSpecFactory
+{
+    private string _fullPath;
+    private string _directory;
+    private TestTargetFramework _outerBuild;
+    private Dictionary<string, ITargetFramework>? _targetFrameworks;
+
+    public TestPackageSpecFactory(string fullPath, Action<TargetFrameworkBuilder> builder)
+        : this(fullPath, System.IO.Path.GetDirectoryName(fullPath) ?? throw new ArgumentException(message: "Could not get directory from path", paramName: nameof(fullPath)), builder)
+    {
+    }
+
+    public TestPackageSpecFactory(string fullPath, string directory, Action<TargetFrameworkBuilder> builder)
+    {
+        _fullPath = string.IsNullOrWhiteSpace(fullPath) ? throw new ArgumentNullException(nameof(fullPath)) : fullPath;
```

Comment:
nit (since this is a test util): I believe we use ArgumentException with a special string for null or whitespace. Could consider at least changing exception type, but I imagine it's low impact.